tests/test_astronomer_namespace_pools.py

```diff
@@ -0,0 +1,241 @@
+from . import supported_k8s_versions, get_containers_by_name
+
+
+@pytest.mark.parametrize(
+    "kube_version",
+    supported_k8s_versions,
+)
+def test_astronomer_namespace_pools_rbac(kube_version):
+    """Test that helm renders astronomer/commander RBAC resources properly when working with namespace pools"""
+
+    # rbacEnabled and clusterRoles and namespacePools set to true, should create Roles and Rolebindings for namespace in Pool
+    # and ignore the cluster role configuration
+    namespaces = ["my-namespace-1", "my-namespace-2"]
+    docs = render_chart(
+        kube_version=kube_version,
+        values={
+            "global": {
+                "rbacEnabled": True,
+                "clusterRoles": True,
+                "features": {
+                    "namespacePools": {
+                        "enabled": True,
+                        "namespaces": {"create": True, "names": namespaces},
+                    },
+                },
+            }
+        },
+        show_only=[
+            "charts/astronomer/templates/commander/commander-role.yaml",
+            "charts/astronomer/templates/commander/commander-rolebinding.yaml",
+        ],
+    )
+
+    assert len(docs) == 4
+
+    # assertions on Role objects
+    for i in range(0, 2):
+        role = docs[i]
+
+        assert role["kind"] == "Role"
+        assert len(role["rules"]) > 0
+        assert role["metadata"]["namespace"] == namespaces[i]
+
+    for i in range(2, 4):
+        role_binding = docs[i]
+
+        expected_subject = {
+            "kind": "ServiceAccount",
+            "name": "release-name-commander",
+            "namespace": "default",
+        }
+        expected_role = {
+            "apiGroup": "rbac.authorization.k8s.io",
+            "kind": "Role",
+            "name": "release-name-commander",
+        }
+
+        assert role_binding["kind"] == "RoleBinding"
+        assert role_binding["metadata"]["namespace"] == namespaces[i - 2]
+        assert role_binding["roleRef"] == expected_role
+        assert role_binding["subjects"][0] == expected_subject
+
+
+@pytest.mark.parametrize(
+    "kube_version",
+    supported_k8s_versions,
+)
+def test_astronomer_namespace_pools_namespaces(kube_version):
+    """Test that Namespaces resources are rendered properly when using namespacePools feature"""
+    # If namespace Pools creation enabled -> create the namespaces
+    namespaces = ["my-namespace-1", "my-namespace-2"]
+    docs = render_chart(
+        kube_version=kube_version,
+        values={
+            "global": {
+                "features": {
+                    "namespacePools": {
+                        "enabled": True,
+                        "namespaces": {"create": True, "names": namespaces},
+                    }
+                }
+            }
+        },
+        show_only=["charts/astronomer/templates/namespaces.yaml"],
+    )
+
+    assert len(docs) == 2
+    for i in range(0, 2):
+        namespace = docs[i]
```

Comment:
```suggestion
    for doc in docs:
```

Reply:
Same here, I need to compare this namespace resource with namespace name in the `namespaces` list variable.

If you want I can change it like you mentioned and use 
```python
assert namespace["metadata"]["name"] in namespaces
```